### PR TITLE
EVG-7952 enqueue a patch

### PR DESCRIPTION
**Pull request by ybrill**
Add an option to the patch action dropdown to enqueue the patch. If enqueuing isn't allowed for the patch (i.e. its patches aren't in mbox format) disable the option.

I'm not sure what to make of the changes to package.json and package-lock.json.

### CODE REVIEW

File: src/components/PatchActionButtons/EnqueuePatch.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        onCancel={hideMenu}
+        okText="Yes"
+        cancelText="Cancel"
+      >
+        <DropdownItem
+          data-cy="enqueue-patch"
+          disabled={disabled || loadingEnqueuePatch}
+          ref={ref}
+        >
+          <Disclaimer>Enqueue patch</Disclaimer>
```
Comment (john-m-liu) [inline src/components/PatchActionButtons/EnqueuePatch.tsx:73]: Similar to my comment on the evergreen pr, I think this label should make it clear what you're enqueuing into. What if we instead label this "add to commit queue?" I checked and that length will be shorter than the current longest label so it won't look awkward

File: cypress/integration/patch/action-buttons.ts [hunk 2]
```diff
@@ -95,4 +95,13 @@ describe("Patch Action Buttons", () => {
...
       .should("exist");
   });
+
+  it("Clicking 'Enqueue Patch' button shows popconfirm with input and banner on success", () => {
+    cy.dataCy("ellipsis-btn").click();
+    cy.dataCy("enqueue-patch").click();
+    cy.get(".ant-btn.ant-btn-primary.ant-btn-sm")
+      .contains("Yes")
+      .click({ force: true });
+    cy.dataCy("banner").should("exist");
```
[the remark this answers is not in the document]
Comment (ybrill) [reply cypress/integration/patch/action-buttons.ts]: done

File: package.json [hunk 3]
```diff
@@ -139,8 +139,8 @@
     "@typescript-eslint/parser": "2.31.0",
     "bugsnag-sourcemaps": "1.3.0",
     "colors": "1.4.0",
-    "cypress": "4.8.0",
-    "eslint": "6.8.0",
+    "cypress": "^4.8.0",
+    "eslint": "^6.8.0",
```
[the remark this answers is not in the document]
Comment (ybrill) [reply package.json]: done

File: src/components/PatchActionButtons/EnqueuePatch.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import { DropdownItem } from "components/ButtonDropdown";
+import { Disclaimer } from "@leafygreen-ui/typography";
+import { useBannerDispatchContext } from "context/banners";
+import {
+  EnqueuePatchMutation,
+  EnqueuePatchMutationVariables,
+} from "gql/generated/types";
+import { ENQUEUE_PATCH } from "gql/mutations";
+import { usePatchAnalytics } from "analytics";
+import { useMutation } from "@apollo/react-hooks";
```
[the remark this answers is not in the document]
Comment (ybrill) [reply src/components/PatchActionButtons/EnqueuePatch.tsx]: done

File: src/components/PatchActionButtons/EnqueuePatch.tsx (hunk 1 above)
Comment (ybrill) [reply src/components/PatchActionButtons/EnqueuePatch.tsx]: done

**Comment by ybrill** [comment]
@john-m-liu I added the option to the patch card's dropdown.
I don't know a way for Cypress to get the particular card I've set up for enqueueing. For now it's just verifying the option is in the list.

**Pull request merged**